File: src/main/java/org/cubectl/identity/auth/JwtProperties.java
package org.cubectl.identity.auth;

import java.time.Duration;

import org.springframework.boot.context.properties.ConfigurationProperties;

@ConfigurationProperties(prefix = "cubectl.jwt")
public record JwtProperties(
        String issuer,
        String secret,
        Duration accessTokenTtl,
        Duration refreshTokenTtl
) {
}

File: src/main/kotlin/org/cubectl/identity/auth/JwtProperties.kt
package org.cubectl.identity.auth

import org.springframework.boot.context.properties.ConfigurationProperties
import java.time.Duration

@ConfigurationProperties(prefix = "cubectl.jwt")
data class JwtProperties(
    val issuer: String = "cubectl-auth",
    val secret: String = "dev-only-change-this-secret-dev-only-change-this-secret",
    val accessTokenTtl: Duration = Duration.ofMinutes(15),
    val refreshTokenTtl: Duration = Duration.ofDays(30),
)
